File: app/src/main/java/com/nruge/iceinfo/model/MenuModels.kt
package com.nruge.iceinfo.model

import kotlinx.serialization.Serializable

@Serializable
data class MenuPageResponse(
    val teaserGroups: List<MenuTeaserGroup> = emptyList(),
    val messages: List<MenuMessage> = emptyList(),
    val declarationGroup: MenuDeclarationGroup? = null
)

@Serializable
data class MenuTeaserGroup(
    val title: String = "",
    val items: List<MenuItem> = emptyList()
)

@Serializable
data class MenuItem(
    val id: Int = 0,
    val title: String = "",
    val subject: String = "",
    val picture: MenuPicture? = null,
    val priceInfo: MenuPriceInfo? = null,
    val declarationBox: MenuDeclarationBox? = null,
    val visible: Boolean = true
) {
    val imageUrl: String get() = picture?.src ?: ""
    val eurPrice: Double? get() = priceInfo?.prices?.firstOrNull { it.currency == "EUR" }?.value
    val declarationKeys: List<String>
        get() = declarationBox?.productMainGroup?.declarationGroups
            ?.flatMap { it.keys } ?: emptyList()
}

@Serializable
data class MenuPicture(val src: String = "")

@Serializable
data class MenuPriceInfo(
    val prices: List<MenuPrice> = emptyList()
)

@Serializable
data class MenuPrice(val currency: String = "", val value: Double = 0.0)

@Serializable
data class MenuDeclarationBox(
    val productMainGroup: MenuProductGroup? = null
)

@Serializable
data class MenuProductGroup(
    val declarationGroups: List<MenuItemDeclarationGroup> = emptyList()
)

@Serializable
data class MenuItemDeclarationGroup(
    val keys: List<String> = emptyList()
)

@Serializable
data class MenuDeclarationGroup(
    val items: List<MenuDeclarationEntry> = emptyList()
)

@Serializable
data class MenuDeclarationEntry(
    val key: String = "",
    val text: String = ""
)

@Serializable
data class MenuMessage(
    val type: String = "",
    val text: String = ""
)

@Serializable
data class AvailabilityItem(
    val ecmId: Int = 0,
    val status: String = "",
    val visible: Boolean = true
)

data class MenuCategory(
    val title: String,
    val items: List<MenuItem>
)

val MENU_CATEGORY_ORDER = listOf(
    "Aktion", "Snacks", "Hauptgerichte", "Frühstück", "Süß & salzig",
    "Suppe", "Kindermenü", "Vegetarisch", "Vegan", "Heißgetränke",
    "Kaltgetränke", "Alkoholische Getränke"
)

File: ios/ICEInfo/Models/MenuModels.swift
import Foundation

struct MenuPageResponse: Codable, Sendable {
    var teaserGroups: [MenuTeaserGroup] = []
    var messages: [MenuMessage] = []
    var declarationGroup: MenuDeclarationGroup? = nil
}

struct MenuTeaserGroup: Codable, Sendable {
    var title: String = ""
    var items: [MenuItem] = []
}

struct MenuItem: Codable, Sendable, Identifiable {
    var id: Int = 0
    var title: String = ""
    var subject: String = ""
    var picture: MenuPicture? = nil
    var priceInfo: MenuPriceInfo? = nil
    var declarationBox: MenuDeclarationBox? = nil
    var visible: Bool = true

    var imageUrl: String { picture?.src ?? "" }
    var eurPrice: Double? { priceInfo?.prices.first { $0.currency == "EUR" }?.value }
    var declarationKeys: [String] {
        declarationBox?.productMainGroup?.declarationGroups.flatMap { $0.keys } ?? []
    }
}

struct MenuPicture: Codable, Sendable {
    var src: String = ""
}

struct MenuPriceInfo: Codable, Sendable {
    var prices: [MenuPrice] = []
}

struct MenuPrice: Codable, Sendable {
    var currency: String = ""
    var value: Double = 0.0
}

struct MenuDeclarationBox: Codable, Sendable {
    var productMainGroup: MenuProductGroup? = nil
}

struct MenuProductGroup: Codable, Sendable {
    var declarationGroups: [MenuItemDeclarationGroup] = []
}

struct MenuItemDeclarationGroup: Codable, Sendable {
    var keys: [String] = []
}

struct MenuDeclarationGroup: Codable, Sendable {
    var items: [MenuDeclarationEntry] = []
}

struct MenuDeclarationEntry: Codable, Sendable {
    var key: String = ""
    var text: String = ""
}

struct MenuMessage: Codable, Sendable {
    var type: String = ""
    var text: String = ""
}

struct AvailabilityItem: Codable, Sendable {
    var ecmId: Int = 0
    var status: String = ""
    var visible: Bool = true
}

struct MenuCategory: Sendable {
    var title: String
    var items: [MenuItem]
}

struct MenuResult: Sendable {
    var categories: [MenuCategory]
    var declarations: [String: String]
}

let menuCategoryOrder = [
    "Aktion", "Snacks", "Hauptgerichte", "Frühstück", "Süß & salzig",
    "Suppe", "Kindermenü", "Vegetarisch", "Vegan", "Heißgetränke",
    "Kaltgetränke", "Alkoholische Getränke"
]
